app/models/concerns/acts_as_linked_list.rb

```diff
@@ -36,28 +36,51 @@ def list_item_name(name)
     #   the new next items.
     #
     # item - The item to be moved.
+    # parent - The item's parent
     # new_position - The hash that contains the new previous and/or new next
     #   item using the keys :prev_item and :next_item, respectively.
     #
-    # Returns true.
-    def move(item, new_position = {})
+    # Returns true when item is moved successfully to the new position.
+    # Returns false when the new position is invalid.
+    def move(item, parent, new_position = {})
       item.class.transaction do
         previous_item = item.send("previous_#{item.class.name.downcase}")
         next_item = item.send("next_#{item.class.name.downcase}")
+        new_previous = new_position[:prev_item]
+        new_next = new_position[:next_item]
+
+        # Validates new position
+        is_valid_new_position = if new_previous.present?
+                                  next_item_of_new_previous = new_previous.send("next_#{item.class.name.downcase}")
+                                  if next_item_of_new_previous
+                                    new_next == next_item_of_new_previous
+                                  else
+                                    new_next.nil?
+                                  end
+                                else
+                                  if parent.items.empty?
+                                    new_next.nil?
+                                  else
+                                    if parent.first_item
+                                      new_next == parent.first_item
+                                    else
+                                      new_next.nil?
+                                    end
+                                  end
+                                end
+        return false unless is_valid_new_position
 
         # Update previous position
         next_item.update_attribute(:previous_id, item.previous_id) if next_item
 
         # Update new position
-        new_previous = new_position[:prev_item]
-        new_next = new_position[:next_item]
+        new_next.update_attribute(:previous_id, item.id) if new_next
```

Comment:
The move method was returning nil when new_next was not present therefore I have moved update new_next here and update new_previous after that to ensure we always return true or false.